docs/hooks/hook_civicrm_alterMailParams.md

```diff
@@ -12,15 +12,39 @@ This hook is called when an email is about to be sent by CiviCRM.
 
 -   $params - the mailing params array
 -   $context - the contexts of the hook call are:
-    -   "civimail" for when sending an email using civimail,
+    -   "civimail" for when sending an email using CiviMail,
+    -   "flexmailer" for when sending an email using CiviMail and
+        [FlexMailer](https://civicrm.org/extensions/flexmailer)
+        (e.g. for Mosaico),
     -   "singleEmail" for when sending a single email,
     -   "messageTemplate" for when sending an email using a message
         template
 
 ## Details
 
 -   $params array fields include: groupName, from, toName, toEmail,
-    subject, cc, bcc, text, html, returnPath, replyTo, headers,
-    attachments (array)
+    subject, cc, bcc, text, html, returnPath, replyTo, headers (array),
+    attachments (array), and possibly others depending on context.
 -   If you want to abort the mail from being sent, set the boolean
-    abortMailSend to true in the params array 
+    abortMailSend to true in the params array
+-   Note that this hook is called twice when sending a message template, once
+    early in the message generation (before tokens are applied, with the context
+    `messageTemplate`) and then later from `CRM_Utils_Mail::send()` with the
+    context `singleEmail`.
+
+
+## Adding custom headers to the email
+
+You can add custom headers by appending to `$params['headers']`. Example:
+
+    $params['headers']['X-My-Header'] = 'my header value';
```

Comment:
```suggestion
    ``` php
    $params['headers']['X-My-Header'] = 'my header value';
    ```
```

Example code should be inside a fenced code block with a language tag specified.